src/components/FullCourt/index.tsx

```diff
@@ -11,10 +11,39 @@ import CourtDimension from "../BasketballCourt/CourtDimension";
 import BorderDimension from "../BasketballCourt/BorderDimension";
 import DashedLine from "../BasketballCourt/DashedLine";
 import useCourt from "@/hooks/useCourt";
+import { IZoomShift } from "@/interfaces/zoomShift";
+import { useStoreSelector } from "@/store/hooks";
+import { centerZoom } from "@/utils/zoomCenterCalculate";
+import { useRef } from "react";
 
 const FullCourt = () => {
   const { courtAreaXLength, courtAreaYLength, borderLength, court, courtStartPoint } = useCourt();
 
+  const zoomScale = useStoreSelector((state) => state.zoomControl.zoomScale);

```

Comment:
It seems like there are lots of duplicated code in different courts, I'm wondering are there any possible way to make these code reusable? It's totally ok if not. Thank you!

Reply:
New changes related to the duplicate code are updated.